test/server/roles.spec.js

```diff
@@ -0,0 +1,265 @@
+const request = require('supertest');
+const assert = require('chai').assert;
+require('babel-register');
+const app = require('../../build/server');
+
+let token;
+
+describe('Role Controller ', () => {
+  beforeEach((done) => {
+    Role.destroy({
+      where: {},
+      truncate: true,
+      cascade: true,
+      restartIdentity: true
+    }).then((error) => {
+      if (!error) {
+        Document
+          .destroy({
+            where: {},
+            truncate: true,
+            cascade: true,
+            restartIdentity: true
+          })
+          .then((err) => {
+            if (!err) {
+              User.destroy({
+                where: {},
+                truncate: true,
+                cascade: true,
+                restartIdentity: true
+              }).then((err) => {
+                if (!err) {
+                  Role.bulkCreate([
+                    {
+                      title: 'administrator'
+                    },
+                    {
+                      title: 'editor'
+                    },
+                    {
+                      title: 'subcriber'
+                    }
+                  ]).then((err) => {
+                    if (!err) {
+                      //
+                    }
+                    done();
+                  });
+                }
+              });
+            }
+          });
+      }
+    });
+  });
+  it('throws a 400 if users try to create an incorrect role', (done) => {
+    request(app)
+      .post('/api/v1/users/')
+      .send({
+        email: 'admin@admin.com',
+        username: 'admin',
+        password: 'edoccabinet',
+        roleId: 1
+      })
+      .expect(201)
+      .end((err, res) => {
+        token = res.body.token;
+        request(app)
+          .post('/api/v1/roles/')
+          .send()
+          .set('Authorization', `${token}`)
+          .set('Accept', 'application/json')
+          .expect('Content-Type', /json/)
+          .expect(400)
+          .end((err, res) => {
+            expect(res.status).to.equal(400);
+          });
+        done();
+      });
+  });
+    it('throws a 201 if admin try to create a new role', (done) => {
+    request(app)
+      .post('/api/v1/users/')
+      .send({
+        email: 'admin@admin.com',
+        username: 'admin',
+        password: 'edoccabinet',
+        roleId: 1
+      })
+      .expect(201)
+      .end((err, res) => {
+        token = res.body.token;
+        request(app)
+          .post('/api/v1/roles/')
+          .send({
+            title: 'publisher'
+          })
+          .set('Authorization', `${token}`)
+          .set('Accept', 'application/json')
+          .expect('Content-Type', /json/)
+          .expect(400)
+          .end((err, res) => {
+            expect(res.status).to.equal(201);
+          });
+        done();
+      });
+  });
+      it('throws a 400 if other roles try to create a new role', (done) => {
+    request(app)
```

Comment:
Should this throw a 403 error instead to show that the other roles are unauthorized?